### ISSUE TITLE
SavedModel Exporting doesn't export ludwig layers with the model

### ISSUE THREAD

**Issue by smiraldr** [ludwig-ai/ludwig#1672]
**Describe the bug**
While using export_savedmodel some ludwig layers are skipped which may result in faulty output when used for inference.

**To Reproduce**

1. Train any bert based model on data
2. Use export_savedmodel function to export 

Logs:
Some layers from the model checkpoint at bert-base-uncased were not used when initializing TFBertModel: ['mlm___cls', 'nsp___cls']
- This IS expected if you are initializing TFBertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing TFBertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
All the layers of TFBertModel were initialized from the model checkpoint at bert-base-uncased.
If your task is similar to the task the model of the checkpoint was trained on, you can already use TFBertModel for predictions without further training.
WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
WARNING:tensorflow:Skipping full serialization of Keras layer <ludwig.modules.reduction_modules.SequenceReducer object at 0x7ef9fdda6160>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <ludwig.modules.reduction_modules.SequenceReducer object at 0x7ef9fdda6160>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <ludwig.modules.reduction_modules.ReduceSum object at 0x7ef9fdda6880>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <ludwig.modules.reduction_modules.ReduceSum object at 0x7ef9fdda6880>, because it is not built.
WARNING:absl:Found untraced functions such as dense_8_layer_call_fn, dense_8_layer_call_and_return_conditional_losses, dense_8_layer_call_fn, dense_8_layer_call_and_return_conditional_losses, dense_8_layer_call_and_return_conditional_losses while saving (showing 5 of 1055). These functions will not be directly callable after loading.
INFO:tensorflow:Assets written to: models/results/BERT_128_712_bert_128_0/export/assets
INFO:tensorflow:Assets written to: models/results/BERT_128_712_bert_128_0/export/assets

### After loading model to tf with following

`model = tf.saved_model.load("models/results/BERT_128_712_bert_128_0/export")`
logs:
WARNING:tensorflow:Unresolved object in checkpoint: (root).output_features.category.decoder_obj.dense.kernel
WARNING:tensorflow:Unresolved object in checkpoint: (root).output_features.category.decoder_obj.dense.kernel
WARNING:tensorflow:Unresolved object in checkpoint: (root).output_features.category.decoder_obj.dense.bias
WARNING:tensorflow:Unresolved object in checkpoint: (root).output_features.category.decoder_obj.dense.bias
WARNING:tensorflow:A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/guide/checkpoint#loading_mechanics for details.
WARNING:tensorflow:A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/guide/checkpoint#loading_mechanics for details.

`print(list(model.signatures.keys()))`
['serving_default']

`print(model.signatures["serving_default"])`
ConcreteFunction signature_wrapper(*, text_input_feature_8_input)
  Args:
    text_input_feature_8_input: int32 Tensor, shape=(None, None)
  Returns:
    {'category_output_feature_8': <1>, 'category_output_feature_8_1': <2>, 'category_output_feature_8_2': <3>}
      <1>: float32 Tensor, shape=(None, 768)
      <2>: float32 Tensor, shape=(None, 28)
      <3>: float32 Tensor, shape=(None, 768)

While my original ludwig input and output signatures were:
"config": {
        "combiner": {
            "type": "concat"
        },
        "input_features": [
            {
                "column": "clean_title",
                "encoder": "bert",
                "level": "word",
                "name": "clean_title",
                "pretrained_model_name_or_path": "bert-base-uncased",
                "proc_column": "clean_title_mZFLky",
                "tied": null,
                "type": "text"
            }
        ],
        "output_features": [
            {
                "column": "category",
                "dependencies": [],
                "loss": {
                    "class_similarities_temperature": 0,
                    "class_weights": 1,
                    "confidence_penalty": 0,
                    "labels_smoothing": 0,
                    "robust_lambda": 0,
                    "type": "softmax_cross_entropy",
                    "weight": 1
                },
                "name": "category",
                "proc_column": "category_mZFLky",
                "reduce_dependencies": "sum",
                "reduce_input": "sum",
                "top_k": 3,
                "type": "category"
            }
        ]

P.S. Everytime I try to export the model to savedmodel format and check the output signature - category_output_feature_8 changes to category_output_feature_{anydifferentnumber}

Environment (please complete the following information):
OS: Ubuntu 18.04
Version
Python version - 3.8.10
Ludwig version - 0.4 tf inference (installed using pip install ludwig)

**Comment by justinxzhao** [comment]
It's not clear that the skipped layers on `tf-legacy`'s `export_savedmodel` would certainly break prediction. I wonder if the layers that tensorflow doesn't trace during `export_savedmodel` means that those layers aren't actually used. The warnings do seem concerning though it's also possible that tensorflow is being extra communicative.

Serializing a Huggingface `TFBertModel` to torchscript with `save_torchscript` on `master` (pytorch) seems to work, with many fewer and much more minor warnings compared to what you're seeing on `tf-legacy`.

Since we're moving to pytorch, I'll close this issue, but @smiraldr feel free to follow up with a new issue if you see exporting issues on `master`.
